xmtp_mls/src/groups/group_permissions.rs

```diff
@@ -1,162 +1,247 @@
 use super::validated_commit::{AggregatedMembershipChange, CommitParticipant, ValidatedCommit};
-
-pub trait Policy {
+use prost::Message;
+use thiserror::Error;
+use xmtp_proto::xmtp::mls::message_contents::{
+    membership_policy::{
+        AndCondition as AndConditionProto, AnyCondition as AnyConditionProto,
+        BasePolicy as BasePolicyProto, Kind as PolicyKindProto,
+    },
+    MembershipPolicy as MembershipPolicyProto, PolicySet as PolicySetProto,
+};
+
+// A trait for policies that can add/remove members and installations for the group
+pub trait MembershipPolicy: std::fmt::Debug {
     fn evaluate(&self, actor: &CommitParticipant, change: &AggregatedMembershipChange) -> bool;
+    fn to_proto(&self) -> MembershipPolicyProto;
+}
+
+#[derive(Debug, Error)]
+pub enum PolicyError {
+    #[error("serialization {0}")]
+    Serialization(#[from] prost::EncodeError),
+    #[error("deserialization {0}")]
+    Deserialization(#[from] prost::DecodeError),
+    #[error("invalid policy")]
+    InvalidPolicy,
 }
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, PartialEq)]
 #[allow(dead_code)]
-pub enum StandardPolicies {
+#[repr(u8)]
+pub enum BasePolicies {
     Allow,
     Deny,
     // Allow if the change only applies to subject installations with the same account address as the actor
     AllowSameMember,
```

Comment:
This is technically only useful for remove installation policies. We could make it error if you try and export one of these policies to a proto, so that it can only be used for one of the hardcoded policies. But that feels a little hacky. For add/remove members applying this policy will effectively be the same as `deny`, so it's probably safe to leave as-is.